GPU/TPCFastTransformation/macro/TPCFastTransformInit.C

```diff
@@ -0,0 +1,159 @@
+    auto z2xBin = v->bvox[o2::tpc::TrackResiduals::VoxZ]; // bin number in z/x 0..4
+
+    auto x = trackResiduals.getX(xBin);             // radius of the pad row
+    auto y2x = trackResiduals.getY2X(xBin, y2xBin); // y/x coordinate of the bin ~-0.15 ... 0.15
+    auto z2x = trackResiduals.getZ2X(z2xBin);       // z/x coordinate of the bin 0.1 .. 0.9
+    float y = x * y2x;
+    float z = x * z2x;
+
+    float correctionX = v->D[o2::tpc::TrackResiduals::ResX];
+    float correctionY = v->D[o2::tpc::TrackResiduals::ResY];
+    float correctionZ = v->D[o2::tpc::TrackResiduals::ResZ];
+    int iRoc = (int)v->bsec;
+    int iRow = (int)xBin;
+
+    map.addCorrectionPoint(iRoc, iRow, y, z, correctionX, correctionY, correctionZ);
+    // cout << iVox << ": sec " << iRoc << " row " << (int)xBin << " x y z " << x << " " << y << " " << z
+    //    << " dx dy dz " << correctionX << " " << correctionY << " " << correctionZ << endl;
+  }
+
+  std::unique_ptr<o2::gpu::TPCFastTransform> fastTransform(helper->create(0));
+  o2::gpu::TPCFastSpaceChargeCorrection& corr = fastTransform->getCorrection();
+
+  // check the difference
+
+  double maxDiff[3] = {0., 0., 0.};
+  double sumDiff[3] = {0., 0., 0.};
+  long nDiff = 0;
+
+  for (int iVox = 0; iVox < tree->GetEntriesFast(); iVox++) {
+    tree->GetEntry(iVox);
+
+    auto xBin = v->bvox[o2::tpc::TrackResiduals::VoxX];   // bin number in x (= pad row)
+    auto y2xBin = v->bvox[o2::tpc::TrackResiduals::VoxF]; // bin number in y/x 0..14
+    auto z2xBin = v->bvox[o2::tpc::TrackResiduals::VoxZ]; // bin number in z/x 0..4
+
+    auto x = trackResiduals.getX(xBin);             // radius of the pad row
+    auto y2x = trackResiduals.getY2X(xBin, y2xBin); // y/x coordinate of the bin ~-0.15 ... 0.15
+    auto z2x = trackResiduals.getZ2X(z2xBin);       // z/x coordinate of the bin 0.1 .. 0.9
+    float y = x * y2x;
+    float z = x * z2x;
+
+    float correctionX = v->D[o2::tpc::TrackResiduals::ResX];
+    float correctionY = v->D[o2::tpc::TrackResiduals::ResY];
+    float correctionZ = v->D[o2::tpc::TrackResiduals::ResZ];
+
+    int iRoc = (int)v->bsec;
+    int iRow = (int)xBin;
+
+    float u, v, cx, cu, cv, cy, cz;
+    geo.convLocalToUV(iRoc, y, z, u, v);
+    corr.getCorrection(iRoc, iRow, u, v, cx, cu, cv);
+    geo.convUVtoLocal(iRoc, u + cu, v + cv, cy, cz);
+    cy -= y;
+    cz -= z;
+    double d[3] = {cx - correctionX,
+                   cy - correctionY,
+                   cz - correctionZ};
+    for (int i = 0; i < 3; i++) {
+      if (fabs(maxDiff[i]) < fabs(d[i])) {
+        maxDiff[i] = d[i];
+      }
+      sumDiff[i] += d[i] * d[i];
+    }
+    nDiff++;
+  }
+  for (int i = 0; i < 3; i++) {
+    sumDiff[i] = sqrt(sumDiff[i]) / nDiff;
+  }
+
+  std::cout << "Max difference in x,y,z :  " << maxDiff[0] << " " << maxDiff[1] << " " << maxDiff[2] << endl;
+  std::cout << "Mean difference in x,y,z : " << sumDiff[0] << " " << sumDiff[1] << " " << sumDiff[2] << endl;
```

Comment:
Even with the toy data this gives tiny mean differences and also very small max differences. It looks very good, thanks a lot!